File: src/main/java/brs/db/sql/VersionedValuesSqlTable.java
package brs.db.sql;

import brs.DependencyProvider;
import brs.db.VersionedValuesTable;
import brs.db.store.DerivedTableManager;
import org.jooq.impl.TableImpl;

public abstract class VersionedValuesSqlTable<T, V> extends ValuesSqlTable<T, V> implements VersionedValuesTable<T, V> {
  VersionedValuesSqlTable(String table, TableImpl<?> tableClass, DbKey.Factory<T> dbKeyFactory, DependencyProvider dp) {
    super(table, tableClass, dbKeyFactory, true, dp);
  }

  @Override
  public final void rollback(int height) {
    VersionedEntitySqlTable.rollback(table, tableClass, heightField, latestField, height, dbKeyFactory);
  }

  @Override
  public final void trim(int height) {
    VersionedEntitySqlTable.trim(tableClass, heightField, height, dbKeyFactory);
  }
}

File: src/main/kotlin/brs/db/sql/VersionedValuesSqlTable.kt
package brs.db.sql

import brs.db.ValuesTable
import brs.entity.DependencyProvider
import org.jooq.Field
import org.jooq.impl.TableImpl

internal abstract class VersionedValuesSqlTable<T, V> internal constructor(
    table: String,
    tableClass: TableImpl<*>,
    heightField: Field<Int>,
    latestField: Field<Boolean>?,
    dbKeyFactory: SqlDbKey.Factory<T>,
    private val dp: DependencyProvider
) : ValuesSqlTable<T, V>(table, tableClass, heightField, latestField, dbKeyFactory, true, dp), ValuesTable<T, V> {
    override fun rollback(height: Int) {
        VersionedEntitySqlTable.rollback(dp, table, tableClass, heightField, latestField, height, dbKeyFactory)
    }

    override fun trim(height: Int) {
        VersionedEntitySqlTable.trim(dp, tableClass, heightField, height, dbKeyFactory)
    }
}
